### PR TITLE
Hpl ledger integration

### PR DESCRIPTION
**Pull request by Jair-SDS**
Hpl Implementation
Edit local data of assets
Get/Add/Edit Subaccounts and Virtual Accounts
Transfer tokens between Sub and Remote Accounts

### CODE REVIEW

File: frontend/pages/home/components/ICRC/SubaccountAction.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            className="flex flex-row justify-center items-center w-7 h-7 bg-SelectRowColor rounded-md cursor-pointer"
+            onClick={() => {
+              setDrawer(DrawerOptionEnum.Enum.RECEIVE);
+            }}
+          >
+            <img src={ArrowBottomLeftIcon} className="w-3 h-3" alt="receive-icon" />
+          </div>
+          <p className="text-md">{t("receive")}</p>
+        </div>
+        {/* <div className="flex flex-col justify-center items-center w-1/3 gap-1">
```
Comment (OliverSDS) [inline frontend/pages/home/components/ICRC/SubaccountAction.tsx:56]: delete

File: frontend/pages/workers/timerWorker.ts [hunk 2]
```diff
@@ -1,17 +1,11 @@
...
-};
-
-// eslint-disable-next-line no-restricted-globals
 const timerCode = () => {
   self.onmessage = () => {
-    self.postMessage(WorkerTrans.TRANSACTIONS);
+    self.postMessage("TRANSACTIONS");
     setInterval(() => {
-      self.postMessage(WorkerTrans.TRANSACTIONS);
+      self.postMessage("TRANSACTIONS");
```
Comment (OliverSDS) [inline frontend/pages/workers/timerWorker.ts:5]: Why did it change from enum to strings?

**Review by OliverSDS — CHANGES_REQUESTED** [review]

File: frontend/pages/contacts/components/addContact.tsx [hunk 3]
```diff
@@ -243,28 +243,27 @@ const AddContact = ({ setAddOpen }: AddContactProps) => {
...
         } else {
-          ids.push(subAccIdx);
+          ids.push(subacc);
         }
         // Adding SubAccountContact to the new contact
-        if (valid) auxNewSub.push({ name: newSa.name.trim(), subaccount_index: subAccIdx });
+        if (valid) auxNewSub.push({ name: newSa.name.trim(), subaccount_index: subacc });
       }
     });
     // Check if valid Subaccounts and Valid prev contact info
```
[the remark this answers is not in the document]
Comment (Jair-SDS) [reply frontend/pages/contacts/components/addContact.tsx:269]: Done!

File: frontend/pages/contacts/components/tableContacts.tsx [hunk 4]
```diff
@@ -242,12 +241,6 @@ const TableContacts = ({
     </table>
   );
```
[the remark this answers is not in the document]
Comment (Jair-SDS) [reply frontend/pages/contacts/components/tableContacts.tsx:243]: Done!

File: frontend/pages/home/components/HPL/DrawerVirtual.tsx [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+      <div className="w-full flex flex-row justify-between items-center mt-12 gap-4">
+        <p className="text-sm text-TextErrorColor text-left">{errMsg}</p>
+        <CustomButton className="min-w-[5rem]" onClick={onSave} size={"small"}>
+          {loading ? <LoadingLoader className="mt-1" /> : <p>{t(selectVt ? "save" : "add")}</p>}
+        </CustomButton>
+      </div>
+    </div>
+  );
+
+  function onClose() {
```
[the remark this answers is not in the document]
Comment (Jair-SDS) [reply frontend/pages/home/components/HPL/DrawerVirtual.tsx:162]: Done!

File: frontend/pages/home/components/ICRC/SubaccountAction.tsx (hunk 1 above)
Comment (Jair-SDS) [reply frontend/pages/home/components/ICRC/SubaccountAction.tsx:56]: Done!

**Pull request merged**